components/dependency_injection/advanced.rst

```diff
@@ -218,3 +218,47 @@ You can change the inner service name if you want to:
             ->addArgument(new Reference('bar.wooz'))
             ->setPublic(false)
             ->setDecoratedService('foo', 'bar.wooz');
+
+Deprecating Services
+--------------------
+
+Once you have made decided to deprecate the use of a service (because it is
+outdated, or you decided not to use and maintain it anymore), you can deprecate
+its definition:
+
+.. configuration-block::
+
+   .. code-block: yaml
+      
+      bar:
+         class: stdClass
+         deprecated: true
+         deprecation_message: The service bar is deprecated. You should stop using it, as it will soon be removed.
+    
+    .. code-block:: xml
+
+        <service id="bar" class="stdClass">
+            <deprecated>The service bar is deprecated. You should stop using it, as it will soon be removed.</deprecated>
+        </service>
+
+    .. code-block:: php
+
+        $container->register('bar', 'stdClass')
+            ->setDeprecated(true, 'The service %service_id% is deprecated. You should stop using it, as it will soon be removed.');
+
+Now, every time a service is created using this deprecated definition will
+trigger a deprecation error, advising you to stop or change your uses of that
+service.
+
+.. note::
+   The message is optionnal, so you can basically set only the status, a
```

Comment:
Actually, the default is what I specified into the code examples, but I can remove it and show it here instead ?